src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,43 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { IconProps } from "@components/Icon/Icon";
+import { Dispatch, ReactElement, SetStateAction } from "react";
+import css from "./Slider.module.css";
+
+export interface IconItem {
+    id: string;
+    icon: ReactElement<IconProps>;
+}
+
+export interface TextItem {
+    id: string;
+    name: string;
+}
+
+export interface SliderProps {
+    items: TextItem[] | IconItem[];
+    activeItemId: string;
+    onChange: Dispatch<SetStateAction<string>>;
+}
+
+const isIconItem = (item: TextItem | IconItem): item is IconItem => (item as IconItem).icon !== undefined;
+
+export default function Slider({ items, activeItemId, onChange }: SliderProps): ReactElement<SliderProps> {
+    const getItemClasses = (id: string) => [css.item, id === activeItemId ? css.active : ""].join(" ");
+
+    return (
+        <ul data-test-id="Slider" className={css.wrapper}>
+            {items.map((item) =>
+                isIconItem(item) ? (
+                    <li key={item.id} onClick={() => onChange(item.id)} className={getItemClasses(item.id)}>
```

Comment:
It is missing the actual selection change (clicking the second item doesn't put the active state on it)

Reply:
The state change will be handled by the parent component so that the active item is also accessible from outside the Slider. I have updated the story to reflect a working example.